etl/processor/processor.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/block-explorer/blob/master/LICENSE.md.
+
+package processor
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+
+	"github.com/insolar/block-explorer/etl/interfaces"
+	"github.com/insolar/block-explorer/etl/models"
+	"github.com/insolar/block-explorer/etl/types"
+)
+
+type Processor struct {
+	JDC          <-chan types.JetDrop
+	TaskC        chan Task
+	TaskCCloseMu sync.Mutex
+	Storage      interfaces.Storage
+	Workers      int
+	active       int32
+}
+
+func NewProcessor(jb interfaces.Transformer, storage interfaces.Storage, workers int) *Processor {
+	if workers < 1 {
+		workers = 1
+	}
+	return &Processor{
+		JDC:          jb.GetJetDropsChannel(),
+		Workers:      workers,
+		TaskCCloseMu: sync.Mutex{},
+		Storage:      storage,
+	}
+
+}
+
+var ErrorAlreadyStarted = errors.New("Already started")
+
+func (p *Processor) Start(ctx context.Context) error {
+	if !atomic.CompareAndSwapInt32(&p.active, 0, 1) {
```

Comment:
use `p.TaskCCloseMu` here too to create new `p.TaskC` before close it with `p.Stop`

Reply:
also, if you use mutex to get/update `p.active` value, you don't need atomic swap/load anymore

Reply:
Yes, but atomic is a paradigm